drivers/counter/counter_nrfx_rtc.c

```diff
@@ -67,175 +68,248 @@ static inline const struct counter_nrfx_config *get_nrfx_config(
 				struct counter_nrfx_config, info);
 }
 
-static int counter_nrfx_start(struct device *dev)
+static int start(struct device *dev)
 {
-	nrfx_rtc_enable(&get_nrfx_config(dev)->rtc);
+	nrf_rtc_task_trigger(get_nrfx_config(dev)->rtc, NRF_RTC_TASK_START);
 
 	return 0;
 }
 
-static int counter_nrfx_stop(struct device *dev)
+static int stop(struct device *dev)
 {
-	nrfx_rtc_disable(&get_nrfx_config(dev)->rtc);
+	nrf_rtc_task_trigger(get_nrfx_config(dev)->rtc, NRF_RTC_TASK_STOP);
+	nrf_rtc_task_trigger(get_nrfx_config(dev)->rtc, NRF_RTC_TASK_CLEAR);
 
 	return 0;
 }
 
-static u32_t counter_nrfx_read(struct device *dev)
+static u32_t read(struct device *dev)
 {
-	return nrfx_rtc_counter_get(&get_nrfx_config(dev)->rtc);
+	return nrf_rtc_counter_get(get_nrfx_config(dev)->rtc);
 }
 
-static int counter_nrfx_set_alarm(struct device *dev, u8_t chan_id,
-				  const struct counter_alarm_cfg *alarm_cfg)
+/* Function calculates distance between to values assuming that one first
+ * argument is in front and that values wrap.
+ */
+static u32_t ticks_sub(u32_t val, u32_t old, u32_t top, bool npow2)
```

Comment:
Could we eliminate npow2 and just test the top value?

Reply:
operations are faster when top is 2^n-1. Alternatively it could be a build time flag (`CONFIG_COUNTER_NRF_RTC_POW2_TOP`). When option is enabled (by default enabled) counter would return error if provided top is not 2^n-1. what do you think?

Reply:
anyway, i will remove npow2 arguement. functions should get inlined anyway.